### PR TITLE
Fix/background activation crash

### PR DESCRIPTION
**Pull request by achocron**
Now we should recommend starting in App constructor

### CODE REVIEW

File: Apps/Contoso.UWP.Puppet/App.xaml.cs [hunk 1]
```diff
@@ -124,5 +127,63 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
...
+        }
+    }
+
+    public class BGTask : IBackgroundTask
+    {
+        public void Run(IBackgroundTaskInstance taskInstance)
+        {
+            System.Diagnostics.Debug.WriteLine("run");
+        }
+        public static BackgroundTaskRegistration RegisterBackgroundTask(string taskEntryPoint,
```
Comment (guperrot) [inline Apps/Contoso.UWP.Puppet/App.xaml.cs]: blank lines missing or extra blank lines in this class

File: Apps/Contoso.UWP.Puppet/App.xaml.cs [hunk 2]
```diff
@@ -124,5 +127,63 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
...
+            Push.PushNotificationReceived += PushNotificationReceivedHandler;
+            MobileCenter.Start("42f4a839-c54c-44da-8072-a2f2a61751b2", typeof(Analytics), typeof(Crashes), typeof(Push));
+        }
+    }
+
+    public class BGTask : IBackgroundTask
+    {
+        public void Run(IBackgroundTaskInstance taskInstance)
+        {
+            System.Diagnostics.Debug.WriteLine("run");
```
Comment (guperrot) [inline Apps/Contoso.UWP.Puppet/App.xaml.cs]: Can we improve those logs?

File: SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/IMobileCenterServicePart.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using Windows.ApplicationModel.Activation;
+
+namespace Microsoft.Azure.Mobile
+{
+    public partial interface IMobileCenterService
+    {
+        /// <summary>
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/IMobileCenterServicePart.cs]: This is a callback here, not a user sdk API.
It needs to be renamed OnLaunched in that interface.

**Comment by achocron** [comment]
Also made a change where we send "unknown" for screen size when it is unavailable because otherwise ingestion rejects the log

File: Apps/Contoso.UWP.Puppet/App.xaml.cs [hunk 4]
```diff
@@ -125,4 +130,48 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
...
+        // Adapted from Microsoft documentation
+        public static BackgroundTaskRegistration RegisterBackgroundTask(string taskEntryPoint,
+            string taskName,
+            IBackgroundTrigger trigger,
+            IBackgroundCondition condition)
+        {
+            //
+            // Check for existing registrations of this background task.
+            //
+
```
Comment (guperrot) [inline Apps/Contoso.UWP.Puppet/App.xaml.cs]: Remove blank line

File: Apps/Contoso.UWP.Puppet/App.xaml.cs [hunk 5]
```diff
@@ -125,4 +130,48 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
...
+            foreach (var cur in BackgroundTaskRegistration.AllTasks)
+            {
+                if (cur.Value.Name == taskName)
+                {
+                    //
+                    // The task is already registered.
```
Comment (guperrot) [inline Apps/Contoso.UWP.Puppet/App.xaml.cs]: Multi line comment for singe line?

File: SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/ApplicationLifecycleHelper.cs [hunk 6]
```diff
@@ -9,6 +15,15 @@ public class ApplicationLifecycleHelper : IApplicationLifecycleHelper
         private static bool _started;
         private static bool _suspended;
 
+        // Singleton instance of ApplicationLifecycleHelper
+        private static ApplicationLifecycleHelper _instance;
+        public static ApplicationLifecycleHelper Instance
+        {
+            get { return _instance ?? (_instance = new ApplicationLifecycleHelper()); }
+            // Setter for testing
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/ApplicationLifecycleHelper.cs]: add blank line

File: SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/StorageAdapter.cs [hunk 7]
```diff
@@ -25,6 +25,12 @@ public StorageAdapter(string databasePath)
             {
                 throw new StorageException(e);
             }
+            catch (NullReferenceException e)
+            {
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/StorageAdapter.cs]: ?

File: Apps/Contoso.UWP.Puppet/App.xaml.cs (hunk 4 above)
Comment (achocron) [inline Apps/Contoso.UWP.Puppet/App.xaml.cs]: That's copy-pasted code from MSDN, I can reformat it

**Pull request merged**